tests/system_test.py

```diff
@@ -154,6 +154,14 @@ def retry_exception(function, timeout=TIMEOUT, delay=.001, max_delay=1, exceptio
     Returns what function returns if it succeeds before timeout.
     Raises last exception raised by function on timeout.
     """
+
+    def default_exception_test(exception: Exception):
+        """Do not permit retry on exceptions usually caused by programmer error. The list might be incomplete."""
+        if isinstance(exception, (SyntaxError, UnboundLocalError, RecursionError, ImportError, TypeError)):
+            raise exception
+
+    exception_test = exception_test or default_exception_test
```

Comment:
It seems to me that if the user passes in an exception_test, we will use it, otherwise we will do the default_exception_test. It seems like a good idea to always call the default_exception_test() irrespective of whether an exception_test is passed in or not. By that I mean that the retry_exception function should fail immediately if the exception thrown is any instance of the exceptions mentioned in default_exception_test which indicates user error.

Reply:
The `exception_test` function has been so far implemented as an allow-list (whitelist) as far as I examined usages. When this is done, then you don't need any default set like this.

Sometimes you might want to retry on `TypeError`, I am thinking, and with your modification this would not be possible because function would throw on TypeError always.

Reply:
oh ok, I always thought of the exception_test as an exception handler function. So, if an exception is thrown inside the retry_exception function it would call exception_test with the exception so you can further analyze/process the thrown exception